quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/SimpleUserAppServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package edu.asu.diging.quadriga.core.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.quadriga.core.citesphere.CitesphereConnector;
+import edu.asu.diging.quadriga.core.data.SimpleUserAppRepository;
+import edu.asu.diging.quadriga.core.model.citesphere.CitesphereAppInfo;
+import edu.asu.diging.quadriga.core.model.users.SimpleUserApp;
+import edu.asu.diging.quadriga.core.service.SimpleUserAppService;
+import edu.asu.diging.simpleusers.core.model.impl.SimpleUser;
+
+/**
+ * This class provides a Service for the repository SimpleUserAppRepository to
+ * apply operations on the SimpleUserApp table in the database
+ * 
+ * @author poojakulkarni
+ *
+ */
+@Service
+public class SimpleUserAppServiceImpl implements SimpleUserAppService {
+
+    @Autowired
+    private CitesphereConnector citesphereConnector;
+
+    @Autowired
+    private SimpleUserAppRepository simpleUserAppRepository;
+
+    /* (non-Javadoc)
+     * @see edu.asu.diging.quadriga.core.service.SimpleUserAppService#save(java.lang.String, java.lang.String)
+     */
+    @Override
+    public SimpleUserApp save(String username, String clientId) {
+        SimpleUserApp simpleUserApp = new SimpleUserApp();
+        simpleUserApp.setId(UUID.randomUUID().toString());
+        simpleUserApp.setUsername(username);
+        simpleUserApp.setAppClientId(clientId);
+        return simpleUserAppRepository.save(simpleUserApp);
+    }
+
+    /* (non-Javadoc)
+     * @see edu.asu.diging.quadriga.core.service.SimpleUserAppService#findByUsername(java.lang.String)
+     */
+    @Override
+    public List<SimpleUserApp> findByUsername(String username) {
+        return simpleUserAppRepository.findByUsername(username);
+    }
+
+    /* (non-Javadoc)
+     * @see edu.asu.diging.quadriga.core.service.SimpleUserAppService#delete(java.lang.String, java.lang.String)
+     */
+    @Override
+    public void delete(String username, String appClientId) {
+        SimpleUserApp userApp = simpleUserAppRepository.findByUsernameAndAppClientId(username, appClientId);
+        if (userApp != null) {
+            simpleUserAppRepository.delete(userApp);
+        }
+    }
+
+    /* (non-Javadoc)
+     * @see edu.asu.diging.quadriga.core.service.SimpleUserAppService#getCitesphereApps(edu.asu.diging.simpleusers.core.model.impl.SimpleUser)
+     */
+    @Override
+    public List<CitesphereAppInfo> getCitesphereApps(SimpleUser user) {
```

Comment:
I would put that into a new manager class that only deals with CitesphereAppInfo objects